lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -949,7 +950,7 @@ using namespace Js;
 
     double JavascriptConversion::ToInteger(double val)
     {
-        if(JavascriptNumber::IsNan(val))
+        if(JavascriptNumber::IsNan(val) || val == -0 || val == 0)
```

Comment:
Please can you use `JavascriptNumber::IsZero(val)` instead of `val == -0 || val == 0` that should work for this check (and is used for it elsewhere)

Also then please remove `JavascriptNumber::IsZero(val)` from the condition on lines 955 and 956.